### PR TITLE
Add delete account feature (#4643)

### LINKED ISSUE #4643: [Templates] Include an option to account deletion.
Templates support account creation but does not appear to include an option to initiate account deletion.

### PR DESCRIPTION
**Pull request by msynk**
this closes #4643

### CODE REVIEW

File: src/Templates/AdminPanel/Bit.AdminPanel/global.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "sdk": {
-        "version": "7.0.304",
+        "version": "7.0.305",
```
Comment (ysmoradi) [inline src/Templates/AdminPanel/Bit.AdminPanel/global.json]: Revert this as it's not related to this PR

File: src/Templates/AdminPanel/Bit.AdminPanel/src/Server/Api/appsettings.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "ConnectionStrings": {
-        "SqlServerConnectionString": "Data Source=.; Initial Catalog=AdminPanelDb;Integrated Security=true;Application Name=AdminPanel;TrustServerCertificate=True;"
+        "SqlServerConnectionString": "Data Source=.\\sqlexpress; Initial Catalog=AdminPanelDb;Integrated Security=true;Application Name=AdminPanel;TrustServerCertificate=True;"
```
Comment (ysmoradi) [inline src/Templates/AdminPanel/Bit.AdminPanel/src/Server/Api/appsettings.json]: Revert sqlexpress

File: src/Templates/AdminPanel/Bit.AdminPanel/src/Shared/Dtos/Account/EditUserDto.cs [hunk 3]
```diff
@@ -13,13 +13,19 @@ public class EditUserDto
...
-            Gender = user.Gender
-        };
+            return Gender?.ToString();
+        }
+        set
+        {
+            if (string.IsNullOrEmpty(value) is false)
+            {
+                Gender = Enum.Parse<Gender>(value);
+            }
```
Comment (ysmoradi) [inline src/Templates/AdminPanel/Bit.AdminPanel/src/Shared/Dtos/Account/EditUserDto.cs:28]:
```cs
if (string.IsNullOrEmpty(value) is false)
   Gender = Enum.Parse<Gender>(value);
else
   Gender = null; // what about else?
```

**Review by ysmoradi — CHANGES_REQUESTED** [review]
plz fix issues

File: src/Templates/AdminPanel/Bit.AdminPanel/src/Shared/Dtos/Account/EditUserDto.cs (hunk 3 above)
Comment (msynk) [reply src/Templates/AdminPanel/Bit.AdminPanel/src/Shared/Dtos/Account/EditUserDto.cs:28]: it's an old code, I just moved it from UserDto. not sure its a valid chagne

File: src/Templates/BlazorDual/Bit.BlazorDual/src/BlazorDual.Shared/Dtos/Account/EditUserDto.cs [hunk 4]
```diff
@@ -13,13 +13,19 @@ public class EditUserDto
...
-            Gender = user.Gender
-        };
+            return Gender?.ToString();
+        }
+        set
+        {
+            if (string.IsNullOrEmpty(value) is false)
+            {
+                Gender = Enum.Parse<Gender>(value);
+            }
```
[the remark this answers is not in the document]
Comment (msynk) [reply src/Templates/BlazorDual/Bit.BlazorDual/src/BlazorDual.Shared/Dtos/Account/EditUserDto.cs:28]: it's an old code, I just moved it from UserDto. not sure its a valid chagne

**Review by ysmoradi — APPROVED** [review]

**Pull request merged**